src/test/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/EnvironmentURLUtilsTest.java

```diff
@@ -0,0 +1,243 @@
+        assertFalse(modifiedJsonNode.toString().contains(LOCALHOST_URL + APPLY_NOC_DECISION_SUBMIT_EVENT_PATH));
+        assertTrue(modifiedJsonNode.toString()
+                .contains(APPLY_NOC_DECISION_SUBMIT_EVENT_DEFAULT_HOST + APPLY_NOC_DECISION_SUBMIT_EVENT_PATH));
+    }
+
+    @Test
+    @SetEnvironmentVariable(key = EnvironmentURLUtils.TEST_STUB_SERVICE_BASE_URL, value = LOCALHOST_URL)
+    void caseEventTestStubServiceBaseUrlUpdated() throws IOException {
+        JsonNode modifiedJsonNode = EnvironmentURLUtils.updateCallBackURLs(caseEventJson, CASE_EVENT);
+
+        // check JSON has been modified
+        assertNotEquals(caseEventJson, modifiedJsonNode);
+
+        // check placeholders are not still present
+        assertFalse(modifiedJsonNode.toString().contains(NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_HOST));
+        assertFalse(modifiedJsonNode.toString().contains(APPLY_NOC_DECISION_SUBMIT_EVENT_HOST));
+
+        // check URLs have been substituted with correct values
+        assertTrue(modifiedJsonNode.toString()
+                .contains(NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_DEFAULT_HOST +
+                          NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_PATH));
+        assertFalse(modifiedJsonNode.toString().contains(LOCALHOST_URL + NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_PATH));
+        assertTrue(modifiedJsonNode.toString().contains(LOCALHOST_URL + APPLY_NOC_DECISION_SUBMIT_EVENT_PATH));
+    }
+
+    @Test
+    @SetEnvironmentVariable(key = EnvironmentURLUtils.MCA_API_BASE_URL, value = LOCALHOST_URL)
+    @SetEnvironmentVariable(key = EnvironmentURLUtils.TEST_STUB_SERVICE_BASE_URL, value = LOCALHOST_URL)
+    void caseEventAllBaseUrlsUpdated() throws IOException {
+        JsonNode modifiedJsonNode = EnvironmentURLUtils.updateCallBackURLs(caseEventJson, CASE_EVENT);
+
+        // check JSON has been modified
+        assertNotEquals(caseEventJson, modifiedJsonNode);
+
+        // check placeholders are not still present
+        assertFalse(modifiedJsonNode.toString().contains(NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_HOST));
+        assertFalse(modifiedJsonNode.toString().contains(APPLY_NOC_DECISION_SUBMIT_EVENT_HOST));
+
+        // check URLs have been substituted with correct values
+        assertTrue(modifiedJsonNode.toString().contains(LOCALHOST_URL + NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_PATH));
+        assertTrue(modifiedJsonNode.toString().contains(LOCALHOST_URL + APPLY_NOC_DECISION_SUBMIT_EVENT_PATH));
+    }
+
+    @Test
+    @SetEnvironmentVariable(key = EnvironmentURLUtils.MCA_API_BASE_URL, value = "httpsMalformed://" + LOCALHOST_URL)
+    void urlNotConvertedOnPreviewEnvironmentTestUrlMalformed()  {
+        Exception exception = assertThrows(MalformedURLException.class,
+                () -> EnvironmentURLUtils.updateCallBackURLs(caseEventJson, CASE_EVENT));
+        assertNotNull(exception);
+        assertTrue(exception.getMessage().contains("unknown protocol"));
+    }
+}
```

Comment:
GitHub linting reporting: `no newline at end of file`.

Reply:
Resolved in https://github.com/hmcts/befta-fw/pull/123/commits/f7fbff8c44e377ce7f637f1419a516c10c184c61